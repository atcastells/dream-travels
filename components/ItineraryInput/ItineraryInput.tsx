import { Itinerary } from "@/features/travels/interfaces";
import { Menu, MenuButton, MenuItem, MenuItems } from "@headlessui/react";
import { ChevronDownIcon } from "@heroicons/react/24/outline";
import classNames from "classnames";
import { Button } from "../Button/Button";
import { Input } from "../Input/Input";
import { TextArea } from "../TextArea/TextArea";

interface ItineraryInputProps {
  itinerary: Itinerary;
  numberOfDays: number;
  index: number;
  onChange: (itinerary: Itinerary, index: number) => void;
}

export const ItineraryInput = ({
  itinerary,
  numberOfDays,
  index,
  onChange,
}: ItineraryInputProps) => {
  return (
    <div className="flex flex-col w-full gap-2 items-end bg-gray-100 rounded-lg p-2">
      <div className="flex flex-row w-full gap-2">
        <Menu>
          <Button className="border border-gray-300 w-20 bg-white flex flex-row justify-center items-center">
            <MenuButton
              as="div"
              className={
                "flex flex-row items-center justify-between w-full p-2"
              }
            >
              <span
                className={classNames("font-normal text-xs", {
                  "text-gray-400": !itinerary.day,
                  "text-black": itinerary.day,
                })}
              >
                {itinerary.day ? itinerary.day : "Day"}
              </span>
              <ChevronDownIcon className="h-3 w-3" />{" "}
            </MenuButton>
          </Button>
          <MenuItems anchor="bottom">
            <div className="flex flex-col w-24 mt-1 bg-white border border-gray-200 rounded-lg">
              {Array.from({ length: numberOfDays }, (_, i) => (
                <MenuItem
                  as={"div"}
                  className="p-2 text-sm cursor-pointer hover:bg-gray-100"
                  onClick={() => onChange({ ...itinerary, day: i + 1 }, index)}
                  key={i}
                >{`${i + 1}`}</MenuItem>
              ))}
            </div>
          </MenuItems>
        </Menu>
        <div className="w-3/4">
          <Input
            name="location"
            id={`location-${index}`}
            value={itinerary.location}
            onChange={(e) =>
              onChange({ ...itinerary, location: e.target.value }, index)
            }
            placeholder="Location"
            className="p-1"
            containerProps={{ className: "flex-1 " }}
          />
        </div>
      </div>
      <div className="w-3/4 flex mr-6">
        <TextArea
          name="description"
          id={`description-${index}`}
          value={itinerary.description}
          onChange={(e) =>
            onChange({ ...itinerary, description: e.target.value }, index)
          }
          placeholder="Description"
          className="p-1"
          containerProps={{ className: "flex-1 " }}
        />
      </div>
    </div>
  );
};
