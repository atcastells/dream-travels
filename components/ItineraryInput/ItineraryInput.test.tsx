import { expect, test } from "@jest/globals";
import { render } from "@testing-library/react";

import { ItineraryInput } from "./ItineraryInput";

class ResizeObserver {
  observe() {}
  unobserve() {}

  disconnect() {}
}

test("ItineraryInput renders correctly", () => {
  window.ResizeObserver = ResizeObserver;
  const { findAllByText, baseElement } = render(
    <ItineraryInput
      itinerary={{ day: 1, location: "Paris", description: "Eiffel Tower" }}
      numberOfDays={1}
      index={0}
      onChange={() => {}}
    />
  );

  expect(findAllByText("1")).toBeTruthy();
  expect(findAllByText("Paris")).toBeTruthy();
  expect(findAllByText("Eiffel Tower")).toBeTruthy();

  expect(baseElement).toMatchSnapshot();
});
