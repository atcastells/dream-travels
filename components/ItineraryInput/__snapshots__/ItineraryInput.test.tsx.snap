// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryInput renders correctly 1`] = `
<body>
  <div>
    <div
      class="flex flex-col w-full gap-2 items-end bg-gray-100 rounded-lg p-2"
    >
      <div
        class="flex flex-row w-full gap-2"
      >
        <button
          class="border border-gray-300 w-20 bg-white flex flex-row justify-center items-center rounded-full px-2"
        >
          <div
            aria-expanded="false"
            aria-haspopup="menu"
            class="flex flex-row items-center justify-between w-full p-2"
            data-headlessui-state=""
            id="headlessui-menu-button-:r2:"
          >
            <span
              class="font-normal text-xs text-black"
            >
              1
            </span>
            <svg
              aria-hidden="true"
              class="h-3 w-3"
              data-slot="icon"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m19.5 8.25-7.5 7.5-7.5-7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
             
          </div>
        </button>
        <div
          class="w-3/4"
        >
          <div
            class="flex flex-col text-sm rounded-full flex-1 "
          >
            <div
              class="flex rounded-full border bg-white border-gray-200 p-1"
            >
              <input
                class="rounded-full bg-white text-black px-2 outline-none w-full p-1"
                id="location-0"
                name="location"
                placeholder="Location"
                type="text"
                value="Paris"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-3/4 flex mr-6"
      >
        <div
          class="flex flex-col text-sm flex-1 "
        >
          <div
            class="flex rounded-xl border bg-white border-gray-200 p-1"
          >
            <textarea
              class="bg-white text-black px-2 outline-none w-full"
              id="description-0"
              name="description"
              placeholder="Description"
              rows="4"
            >
              Eiffel Tower
            </textarea>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
